qiskit/ignis/characterization/calibrations/ibmq_utils.py

```diff
@@ -0,0 +1,207 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Utility functions for calibrating IBMQ Devices
+"""
+
+from collections import defaultdict
+import numpy as np
+from scipy.optimize import least_squares
+from qiskit.pulse import SamplePulse, pulse_lib, Schedule, FrameChange
+from qiskit.pulse.schedule import ParameterizedSchedule
+from qiskit.pulse.commands import FrameChangeInstruction
+
+
+def _fit_drag_func(duration, amp, sigma, beta, exp_samples):
+    """
+    Helper function to compare a drag pulse to samples from
+    the experiment
+
+    Args:
+        duration: pulse duration
+        amp: gauss amp
+        sigma: gauss sigma
+        beta: drag amp
+        exp_samples: the experiment pulse, split into real and imag
+
+    Returns:
+        difference between the drag and experimental samples
+
+    """
+
+    fit_pulse = pulse_lib.drag(duration=duration, amp=amp, sigma=sigma,
+                               beta=beta).samples*np.exp(-1j*np.pi/2)
+
+    return np.concatenate((fit_pulse.real, fit_pulse.imag))-exp_samples
+
+
+def get_single_q_pulse(cmd_def, qubits):
+    """
+    Get the DRAG parameters for the single qubit pulse
+
+    Args:
+        cmd_def: CmdDef object for the device
+        qubits: list of qubits to extract the parameters
+
+    Returns:
+        List of dictionaries with the parameters for the DRAG
+
+    TODO:
+        Deprecated once parameterized pulses are supported
+    """
+
+    fit_params = []
+
+    for q in qubits:
+
+        # get u2 command
+        u2_q = cmd_def.get('u2', qubits=q, P0=1, P1=1)
+
+        for instr in u2_q.instructions:
+            if isinstance(instr[1].command, SamplePulse):
+
+                pulse_samples = instr[1].command.samples
+                pulse_dur = len(pulse_samples)
+                pulse_max = np.max(np.abs(pulse_samples))
+                pulse_samples = np.concatenate((pulse_samples.real,
+                                                pulse_samples.imag))
+                break
+
+        # fit a drag pulse
+        def fit_func_2(x):
+            return _fit_drag_func(pulse_dur, x[0],
+                                  x[1], x[2], pulse_samples)
+
+        opt_result = least_squares(fit_func_2, [pulse_max, pulse_dur/4,
+                                                pulse_max/5])
+
+        fit_params.append({'amp': opt_result.x[0],
+                           'duration': pulse_dur,
+                           'sigma': opt_result.x[1],
+                           'beta': opt_result.x[2]})
+
+    return fit_params
+
+
+def update_u_gates(drag_params, pi2_pulse_schedules=None,
+                   qubits=None, cmd_def=None, system=None):
```

Comment:
You changed `system` -> `drives` in the `rabi_scheduled` function to pass in a list of DriveChannels.. so this should probably be changed as well for consistency.